RecoPixelVertexing/PixelTriplets/interface/HitTripletGeneratorFromPairAndLayers.h

```diff
@@ -33,6 +33,17 @@ class HitTripletGeneratorFromPairAndLayers {
                             const edm::Event & ev, const edm::EventSetup& es,
                             const SeedingLayerSetsHits::SeedingLayerSet& pairLayers,
                             const std::vector<SeedingLayerSetsHits::SeedingLayer>& thirdLayers) = 0;
+
+    virtual void hitTriplets(
+	const TrackingRegion& region, 
+	OrderedHitTriplets & result,
+	const edm::EventSetup & es,
+	const HitDoublets & doublets,
+	const RecHitsSortedInPhi ** thirdHitMap,
+	const std::vector<const DetLayer *> & thirdLayerDetLayer,
+	const int nThirdLayers)
+	{};
```

Comment:
Anshul and I agree to make the new function
HitTripletGeneratorFromPairAndLayers::hitTriplets purely virtual
=> there will be an error on compile time if there is no
implementation in one of the leaf classes

For the leaf classes that are currently not used by FastSim we'll
implement the hitTriplets function to throw an error.
In this way someone who configures FastSim to use one of those leaf
classes will become aware that there is no implementation ready.



On Mon, Apr 18, 2016 at 10:56 AM, Slava Krutelyov
<notifications@github.com> wrote:
> In
> RecoPixelVertexing/PixelTriplets/interface/HitTripletGeneratorFromPairAndLayers.h:
>
>> @@ -33,6 +33,17 @@ class HitTripletGeneratorFromPairAndLayers {
>>                              const edm::Event & ev, const edm::EventSetup&
>> es,
>>                              const SeedingLayerSetsHits::SeedingLayerSet&
>> pairLayers,
>>                              const
>> std::vector<SeedingLayerSetsHits::SeedingLayer>& thirdLayers) = 0;
>> +
>> +    virtual void hitTriplets(
>> +	const TrackingRegion& region,
>> +	OrderedHitTriplets & result,
>> +	const edm::EventSetup & es,
>> +	const HitDoublets & doublets,
>> +	const RecHitsSortedInPhi ** thirdHitMap,
>> +	const std::vector<const DetLayer *> & thirdLayerDetLayer,
>> +	const int nThirdLayers)
>> +	{};
>
> why exception, runtime error, instead of compile time?
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub